First API Test for the the Frontend for Backends Desktop microservice (webagg)

augustocristian opened the pull request:
Includes:
- Base Class that manages authentication using OAuth2 with password-based authentication to obtain the token.
- Test case utilizing the aforementioned token to access various endpoints of the webagg microservice.
- Consolidated BaseAPIClass and BaseLogged to use and declare same variables
- Cleaned and improved the code, solved several smells reported by Sonarlint+IntelliJ static analysis

augustocristian commented:
This PR has the same problem  than #56, I am working on it but in the separate branch: https://github.com/giis-uniovi/retorch-st-eShopContainers/tree/ft_flakyTJobc

javiertuya commented:
@augustocristian Hold this PR until #56 is solved to do not add new features while the system is not stable

augustocristian commented:
Now that the SUT is stable, it might be a good idea to continue with this branch, @javiertuya, and finish implementing the API and the end-to-end test cases?

javiertuya commented:
@augustocristian Maybe, at least, finish this PR to avoid stalled PRs

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/common/BaseAPIClass.java:
```diff
@@ -0,0 +1,111 @@
+import java.util.List;
+import java.util.Properties;
+
+public class BaseAPIClass {
```
I miss some comment to reflect the nature of this class.
Check consistency between web tests and api tests.

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/common/BaseAPIClass.java:
```diff
@@ -0,0 +1,111 @@
+
+        // Add parameters for authentication
+        List<BasicNameValuePair> params = new ArrayList<>(3);
+        params.add(new BasicNameValuePair("grant_type", "password"));
+        params.add(new BasicNameValuePair("client_id", "testalice"));
+        params.add(new BasicNameValuePair("username", "alice"));
+        params.add(new BasicNameValuePair("password", "Pass123$"));
+        params.add(new BasicNameValuePair("client_secret", "secret"));
+        params.add(new BasicNameValuePair("scope", "basket"));
```
Hardcoded data should be in a properties file. Check consistency between web and api tests

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/common/BaseAPIClass.java:
```diff
@@ -0,0 +1,111 @@
+        params.add(new BasicNameValuePair("client_secret", "secret"));
+        params.add(new BasicNameValuePair("scope", "basket"));
+
+        // Set parameters for HTTP request
```
This and below: Every statement has tree lines: a blank, a trivial comment and the line...

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/model/Order.java:
```diff
@@ -0,0 +1,180 @@
+package giis.eshopcontainers.e2e.functional.model;
+import java.util.List;
+
+
+
```
Again extra blank spaces!!

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+import java.io.IOException;
+import java.util.List;
+
+public class WebAggAPITests extends BaseAPIClass {
```
I dont get the meaning of WebAgg, could you chose a better name? and some comment here would be useful.

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+    @Test
+    @DisplayName("testAddProductsBasketWebAgg")
+    void testAddProductsBasket() throws IOException, InterruptedException {
+        // Initialize Gson for JSON serialization/deserialization
```
Trivial comment?

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+        // Initialize Gson for JSON serialization/deserialization
+        Gson gson = new Gson();
+
+        addItemsToBasket();
```
What is this? the test data setup? the test execution? Here is where a comment would be useful

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+        Assertions.assertEquals(12, listitems.get(1).getUnitPrice());
+    }
+
+    // Method to add items to the basket
```
Trivial comment

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+        // Create HTTP client and POST request
+        HttpClient httpclient = HttpClients.createDefault();
+        String requestURL ="http://webshoppingagg_" + tJobName + ":80/api/v1/Basket";
+        //requestURL = "http://localhost:5016//api/v1/Basket";
```
Remove

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/tests/WebAggAPITests.java:
```diff
@@ -0,0 +1,142 @@
+        log.debug("Performing the request");
+        // Execute HTTP request and return response
+        return httpclient.execute(httpPost, responseHandler);
+    }
```
This method and the next have the same problem as the base class, there are many trivial comments

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/model/Order.java:
```diff
@@ -1,9 +1,9 @@
 
+import java.util.List;
 
 public class Order {
```
The review didn't include restructuring the models. Get them back to their state before the review.

NOTE: Never add unrelated changes after a revision because this makes extremely difficult the review process. It is good to improve the code, but this must be handled in a separate PR.

javiertuya replied:
I don't see the changes, the only change since last review is this:
![image](https://github.com/giis-uniovi/retorch-st-eShopContainers/assets/10879637/c197a067-bfbb-49e5-af42-dc02e656355a)

augustocristian replied:
For some reason my IDE reverted to the wrong commit (the previous one instead the original). Sorry for the inconveniences, now the model files are in the original state (the first commit)

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/model/Order.java:
Models have not been reverted

javiertuya commented on src/test/java/giis/eshopcontainers/e2e/functional/model/OrderItem.java:
```diff
@@ -11,36 +11,49 @@ public class OrderItem {
     public void setPictureUrl(String pictureUrl) {
         this.pictureUrl = pictureUrl;
     }
+// Agregar getters y setters
```
Why this comment?

javiertuya reviewed (changes requested):
Additionally, this branch is behind main

augustocristian commented:
Branch updated with the lastest combined-dependency update. Removed comments of the restored code, and removed blankspaces in the header and end of the class. Test passing and ready to merge @javiertuya
